packages/@react-spectrum/slider/src/SliderBase.tsx

```diff
@@ -0,0 +1,197 @@
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {BaseSliderProps, SpectrumSliderTicksBase} from '@react-types/slider';
+import {classNames, useStyleProps} from '@react-spectrum/utils';
+import {LabelPosition, Orientation, StyleProps, ValueBase} from '@react-types/shared';
+import React, {CSSProperties, HTMLAttributes, MutableRefObject, ReactNode, useRef} from 'react';
+import {SliderState, useSliderState} from '@react-stately/slider';
+import styles from '@adobe/spectrum-css-temp/components/slider/vars.css';
+import {useLocale, useNumberFormatter} from '@react-aria/i18n';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useSlider, useSliderThumb} from '@react-aria/slider';
+
+export interface SliderBaseChildArguments {
+  inputRefs: MutableRefObject<undefined>[],
+  thumbProps: HTMLAttributes<HTMLElement>[],
+  inputProps: HTMLAttributes<HTMLElement>[],
+  ticks: ReactNode,
+  state: SliderState
+}
+
+export interface SliderBaseProps extends BaseSliderProps, ValueBase<number[]>, SpectrumSliderTicksBase, StyleProps {
+  children: (SliderBaseChildArguments) => ReactNode,
+  formatOptions?: Intl.NumberFormatOptions,
+  classes?: string[] | Object,
+  style?: CSSProperties,
+  count: 1 | 2,
+  // SpectrumBarSliderBase:
+  orientation?: Orientation,
+  labelPosition?: LabelPosition,
+  valueLabel?: ReactNode,
+  showValueLabel?: boolean
+}
+
+function SliderBase(props: SliderBaseProps) {
+  props = useProviderProps(props);
+  let {
+    tickCount, showTickLabels, tickLabels, isDisabled, count,
+    children, classes, style,
+    labelPosition = 'top', valueLabel, showValueLabel = !!props.label,
+    formatOptions,
+    ...otherProps
+  } = props;
+
+  let {styleProps} = useStyleProps(otherProps);
+  let {direction} = useLocale();
+
+  // Assumes that DEFAULT_MIN_VALUE and DEFAULT_MAX_VALUE are both positive, this value needs to be passed to useSliderState, so
+  // getThumbMinValue/getThumbMaxValue cannot be used here.
+  // `Math.abs(Math.sign(a) - Math.sign(b)) === 2` is true if the values have a different sign and neither is null.
+  let alwaysDisplaySign = props.minValue != null && props.maxValue != null && Math.abs(Math.sign(props.minValue) - Math.sign(props.maxValue)) === 2;
+  if (alwaysDisplaySign) {
+    if (formatOptions != null) {
+      if (!('signDisplay' in formatOptions)) {
+        // @ts-ignore
+        formatOptions.signDisplay = 'exceptZero';
```

Comment:
We probably shouldn't mutate the formatOptions here since it comes from props. Maybe spread into a new object?